### ISSUE TITLE
Az appconfig kv set error if value containing close bracket

### ISSUE THREAD

**Issue by cakriwut** [Azure/azure-cli#16431]
> ### `az feedback` auto-generates most of the information requested below, as of CLI version 2.0.62

**Describe the bug**

Updating/Creating with value that contains ")" close bracket will generate error.
Example, when running following command   
      'az appconfig kv set --value ")test" ' 
it will return error message:  **test** was unexpected at this time. 

**To Reproduce**

Run 'az appconfig kv set --value ' , and set the value to any string that contains ).

**Expected behavior**

The command should insert the value without error.

**Environment summary**

Windows 10
Az cli version 
{
  "azure-cli": "2.17.1",
  "azure-cli-core": "2.17.1",
  "azure-cli-telemetry": "1.0.6",
  "extensions": {
    "azure-firewall": "0.8.0",
    "ip-group": "0.1.2",
    "logic": "0.1.2"
  }
}

**Additional context**

**Comment by yonzhan** [comment]
appconfig

**Comment by cakriwut** [comment]
Possibly the problem in the windows cmd shell.

Command:  az appconfig kv set -n appconfig --key TestKEY --value ")TestValue" --label TEST --yes -y --content-type application/json 

Will be translated to following cmd call

"C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\wbin\\..\python.exe" -IBm azure.cli appconfig kv set -n appconfig --key TestKEY --value )TestValue --label TEST --yes -y --content-type application/json


The close bracket value are not written inside double quotes. I think instead of )TestValue , it should be ")TestValue"  - with double quotes like this : 

"C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\wbin\\..\python.exe" -IBm azure.cli appconfig kv set -n appconfig --key TestKEY --value **")TestValue"**  --label TEST --yes -y --content-type application/json

**Comment by cakriwut** [comment]
I can confirm the double quotes solve the problem. 
I hope the team can make this simple changes. 

> Possibly the problem in the windows cmd shell.
> 
> Command: az appconfig kv set -n appconfig --key TestKEY --value ")TestValue" --label TEST --yes -y --content-type application/json
> 
> Will be translated to following cmd call
> 
> "C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\wbin\..\python.exe" -IBm azure.cli appconfig kv set -n appconfig --key TestKEY --value )TestValue --label TEST --yes -y --content-type application/json
> 
> The close bracket value are not written inside double quotes. I think instead of )TestValue , it should be ")TestValue" - with double quotes like this :
> 
> "C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\wbin\..\python.exe" -IBm azure.cli appconfig kv set -n appconfig --key TestKEY --value **")TestValue"** --label TEST --yes -y --content-type application/json

**Comment by avanigupta** [comment]
Hi @cakriwut, if you are trying to create a key-value with JSON content type (`--content-type application/json`), the value must be an escaped JSON string. Please see this document for more info on JSON key-values in AppConfig: https://docs.microsoft.com/en-us/azure/azure-app-configuration/howto-leverage-json-content-type#create-json-key-values-using-azure-cli

**Comment by cakriwut** [comment]
Hi @avanigupta , it doesn't help.  
az appconfig kv set -n appconfig --key TestKEY --value ")TestValue" --label TEST --yes -y 

Please try on following environment.
1. Windows Powershell => ERROR.  
2. Windows CMD  => OK

Problem
When running az cli command from Powershell, it seems invoking "phyton.exe" azure.cli  {args}. However the arguments are not quoted.

**Comment by avanigupta** [comment]
@cakriwut, there is a known issue with parsing quotes in Powershell. [This document](https://github.com/Azure/azure-cli/blob/dev/doc/quoting-issues-with-powershell.md) can better explain the error you're seeing. 
To prevent this error, you may use stop-parsing symbol `--%` between `az` and arguments.

**Comment by anirugu** [comment]
Today while I using azure static app, I write this command

```
az staticwebapp appsettings set -n anirudhagupta4september1040 --setting-names MONGO_CONNECTION_STRING="commentedforprivacy/?retryWrites=true&w=majority"
```
I am getting error like ''w' is not recognized as an internal or external command,
operable program or batch file.' 

@avanigupta  @cakriwut 

Any idea to make it work. I tried both double quote and (""). None of both is solving the issue. I have reported it on static-app cli as well https://github.com/Azure/static-web-apps-cli/issues/551
